File: benchmarks/oopsla25/subjects/203.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f203(const char* s) {
    int len = strlen(s);
    const char* t = strndup(s, len - 1);
    int i = 0;
    while (i < len - 1) {
        assert(t[i] == 'a');
        i++;
    }
    assert(s[len - 1] != 'a');
    free((void*)t);
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f203(my_string);
  return 0;
}

File: benchmarks/oopsla25/subjects/203.py
#!/usr/bin/env python3

import sys

def f203(s: str):
  t = s[0:len(s)-1]
  i = 0
  while i < len(t):
    assert t[i] == "a"
    i += 1
  assert s[len(s)-1] != "a"

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f203(my_string)
